compiler/il/OMRNode.cpp

```diff
@@ -3679,9 +3679,9 @@ OMR::Node::exceptionsRaised()
       default:
        if (node->getOpCode().isCall() && !node->isOSRFearPointHelperCall())
             {
-            possibleExceptions |= TR::Block::CanCatchOSR;
-            if (node->getSymbolReference()->canGCandExcept()
-               )
+            if (!((TR::MethodSymbol*)node->getSymbolReference()->getSymbol())->functionCallDoesNotYieldOSR())
+               possibleExceptions |= TR::Block::CanCatchOSR;
+            if (!node->isPureCall() && node->getSymbolReference()->canGCandExcept())
```

Comment:
@r30shah is there a case for checking isPureCall inside canGCAndExcept ? Just curious..

Reply:
While making changes, I did look at the query `canGCandExcept` , I tried to see if there are places in code-base where it checks if the recognized method is pure function and sets that flag to `false` explicitly, I could not find one. Also skimmed through the place where we create a method symbol, I do not see it checks if the recognized method is Pure call[1]. As I was not sure if the flag is guaranteed to set to `false` for Pure calls, I opted for conservative approach.

[1]. https://github.com/eclipse/omr/blob/7f70fe94e3c03c8db0424bc9a000e5134cedb2fd/compiler/compile/OMRSymbolReferenceTable.cpp#L1419

Reply:
Okay I think that's fine for this PR.